library/src/armory/model/object_detection/object_detector.py

```diff
@@ -81,28 +83,28 @@ def _filter_predictions(self, preds):
         for pred in preds:
             # Filter based on score shreshold, if configured
             if self.score_threshold is not None:
-                print(f"started with {len(pred['boxes'])} boxes before score threshold")
+                log.info(
+                    f"started with {len(pred['boxes'])} boxes before score threshold"
+                )
                 keep = pred["scores"] > self.score_threshold
                 pred["boxes"] = pred["boxes"][keep]
                 pred["scores"] = pred["scores"][keep]
                 pred["labels"] = pred["labels"][keep]
-                print(f"keeping {len(pred['boxes'])} boxes")
+                log.info(f"keeping {len(pred['boxes'])} boxes")
             # Perform non-maximum suppression, if configured
             if self.iou_threshold is not None:
-                print(f"started with {len(pred['boxes'])} boxes before iou threshold")
+                log.info(
+                    f"started with {len(pred['boxes'])} boxes before iou threshold"
+                )
                 keep = torchvision.ops.nms(
                     boxes=to_torch(pred["boxes"]),
                     scores=to_torch(pred["scores"]),
                     iou_threshold=self.iou_threshold,
                 )
-                single = len(keep) == 1
-                boxes = pred["boxes"][keep]
-                scores = pred["scores"][keep]
-                labels = pred["labels"][keep]
-                pred["boxes"] = [boxes] if single else boxes
-                pred["scores"] = [scores] if single else scores
-                pred["labels"] = [labels] if single else labels
-                print(f"keeping {len(pred['boxes'])} boxes")
+                pred["boxes"] = pred["boxes"][keep]
```

Comment:
I don't remember what model I was using when I ran into this case, but if the `pred` arrays are NumPy arrays (it isn't necessarily required that they be PyTorch Tensors in this class), this produces undesired results when only one box remains after NMS filtering because NumPy likes to automatically unwrap single-element arrays for some reason (Matt has heard me complain about this numerous times).

```python
>>> scores = np.array([0.996, 0.554, 0.314, 0.159])
>>> keep = torch.tensor([0, 1])  # output from torchvision.ops.nms
>>> scores[keep]
array([0.996, 0.554])
>>> keep = torch.tensor([0])
>>> scores[keep]
0.996
>>> scores[keep.numpy()] # but this works for some reason
array([0.996])

```

Reply:
Your last evaluation suggests another solution -- not mixing torch and numpy indexing. We can convert the `keep` indices to a `numpy` array if the predictions are also `numpy` arrays as below.
```               
                keep = torchvision.ops.nms(
                    boxes=to_torch(pred["boxes"]),
                    scores=to_torch(pred["scores"]),
                    iou_threshold=self.iou_threshold,
                )
                if isinstance(pred["scores"], np.ndarray):
                    keep = to_numpy(keep)
                pred["boxes"] = pred["boxes"][keep]
                pred["scores"] = pred["scores"][keep]
                pred["labels"] = pred["labels"][keep]
```

Reply:
Yeah, that approach looks pretty clean to me.